Templates/GitHubCommenter-CSharp/run.csx

```diff
@@ -0,0 +1,50 @@
+// Please follow the link to get information on GitHub authentication   
+// https://developer.github.com/v3/oauth/
+
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+
+public static async void Run(dynamic payload, TraceWriter log)
+{
+    if (payload.action != "opened")
+    {
+        return;
+    }
+
+    string comment = "{ \"body\": \"Thank you for your contribution, We will get to it shortly\" }";
+    string label = "[ \"bug\" ]";
+    
+    if (payload.issue != null)
+    {
+        log.Info($"{payload.issue.user.login} posted an issue #{payload.issue.number}:{payload.issue.title}");
+
+        //Post a comment 
+        await SendGitHubRequest(payload.issue.comments_url.ToString(), comment);
+
+        //Add a label
+        await SendGitHubRequest($"{payload.issue.url.ToString()}/labels", label);
+    }
+
+    if (payload.pull_request != null)
+    {
+        log.Info($"{payload.pull_request.user.login} submitted pull request #{payload.pull_request.number}:{payload.pull_request.title}");
+
+        // posting a comment
+        await SendGitHubRequest(payload.pull_request.comments_url.ToString(), comment);
+    }
+}
+
+public static async Task SendGitHubRequest(string url, string requestBody)
+{
+    using (var client = new HttpClient())
+    {
+        client.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("username", "version"));
+
+        // Add the github access token as an app setting
+        client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic", Environment.GetEnvironmentVariable("GITHUB_CREDENTIALS"));
```

Comment:
Would probably be helpful to enhance the comment since this expects the value to be the Base64 encoded string used for basic auth.